gammapy/image/models/core.py

```diff
@@ -110,6 +111,85 @@ def evaluate(lon, lat, lon_0, lat_0, sigma):
         return norm * np.exp(exponent)
 
 
+class SkyElongatedGaussian(SkySpatialModel):
+    r"""Two-dimensional elongated Gaussian model (defined on the plane).
+
+    .. math::
+
+             f(\text{lon}, \text{lat}) = \frac{1}{2\pi\sigma_{\text{lon}}\sigma_{\text{lat}}} e^{-a\left(\text{lon} - \text{lon}_{0}\right)^{2}  -b\left(\text{lon} - \text{lon}_{0}\right)\left(\text{lat} - \text{lat}\right)  -c\left(\text{lat} - \text{lat}_{0}\right)^{2}}
+
+    Using the following definitions:
+
+            a = \left(\frac{\cos^{2}{\left (\theta \right )}}{2 \sigma_{\text{lon}}^{2}} +
+            \frac{\sin^{2}{\left (\theta \right )}}{2 \sigma_{\text{lat}}^{2}}\right)
+
+            b = \left(\frac{\sin{\left (2 \theta \right )}}{2 \sigma_{\text{lon}}^{2}} -
+            \frac{\sin{\left (2 \theta \right )}}{2 \sigma_{\text{lat}}^{2}}\right)
+
+            c = \left(\frac{\sin^{2}{\left (\theta \right )}}{2 \sigma_{\text{lon}}^{2}} +
+            \frac{\cos^{2}{\left (\theta \right )}}{2 \sigma_{\text{lat}}^{2}}\right)
+
+    where :math:`\sigma_{\text{lon}}` and :math:`\sigma_{\text{lat}}` are the 68% containment semi-axes of the gaussian before rotation, and :math:`\theta` is the rotation angle (with respect to the constant latitude axis).
+
+    Parameters
+    ----------
+    lon_0 : `~astropy.coordinates.Longitude`
+        :math:`lon_0`
+    lat_0 : `~astropy.coordinates.Latitude`
+        :math:`lat_0`
+    sigma_lon : `~astropy.coordinates.Angle`
+        :math:`\sigma_{\text{lon}}`
+    sigma_lat : `~astropy.coordinates.Angle`
+        :math:`\sigma_{\text{lat}}`
+    theta : `~astropy.coordinates.Angle`
+        :math:`\theta}`
+    """
+
+    def __init__(self, lon_0, lat_0, sigma_lon, sigma_lat, theta):
+        self.parameters = Parameters(
+            [
+                Parameter("lon_0", Longitude(lon_0)),
+                Parameter("lat_0", Latitude(lat_0)),
+                Parameter("sigma_lon", Angle(sigma_lon), min=0),
+                Parameter("sigma_lat", Angle(sigma_lat), min=0),
+                Parameter("theta", Angle(theta)),
+            ]
+        )
+
+    @staticmethod
+    def evaluate(lon, lat, lon_0, lat_0, sigma_lon, sigma_lat, theta):
+        """Evaluate the model (static function)."""
+
+        dlon1 = np.abs(lon - lon_0)
+        dlon2 = 360.0 * u.deg - np.abs(lon - lon_0)
+        lon_sep = Angle(np.where(dlon1 <= dlon2, dlon1, dlon2) * u.deg)
+
+        dlat1 = np.abs(lat - lat_0)
+        dlat2 = np.abs(lat) + np.abs(lat_0)
+        dlat3 = 180.0 * u.deg - (np.abs(lat) + np.abs(lat_0))
+        lat_sep = Angle(
+            np.where(
+                np.logical_and(dlat1 <= dlat2, dlat1 <= dlat3),
+                dlat1,
+                np.where(dlat2 <= dlat3, dlat2, dlat3),
+            )
+            * u.deg
+        )
+
```

Comment:
I think the computation of `lon_sep` and `lat_sep` could be much simplified by using `angular_separation`, as we do in the other models.

Reply:
Thank you for your suggestions! I am in the process of implementing them. About the use of angular_separation, that was my original idea as well. However, as the gaussian is not normalized on the sphere, using distances computed on the sphere (line angular_separation does) seems wrong. @registerrier maybe you could add on this?

Reply:
You need to decide whether the Gaussian should be "on the sphere" or in the planar approximation first. 

I put a long issue description summarising what others did previously and a recommendation what to put in Gammapy in #1663 . Please read that and then agree (on Github or have a short call with @registerrier or @adonath) first. I think any choices you make are OK (tradeoff in accuracy vs speed). I'm unsubscribing now.

Reply:
Thanks @cdeil! I agree we have to decide whether our models should be "on the sphere" or in planar approximation. I'll take a look at the references you put in #1663 as well.

Reply:
I think we can have both but it should be clear what each of them are. For most use cases, a gaussian on the plane is good enough. It is important though that the model appears as a function on the plane e.g.: `PlaneGaussian` vs `SkyGaussian` or better `SphereGaussian`.